app/models/user.rb

```diff
@@ -530,4 +536,14 @@ def remove_stale_from_autocomplete
     self.class.remove_from_autocomplete(self.autocomplete_search_string_was, self.autocomplete_prefixes, self.autocomplete_value_was)
   end
 
+  def username_is_not_recently_changed
+    change_interval_days = ArchiveConfig.USER_RENAME_LIMIT_DAYS
+    return unless renamed_at && change_interval_days.days.ago <= renamed_at
```

Comment:
It doesn't look like `USER_RENAME_LIMIT` is used in this calculation, only in the error message. So if you did set it to 2, you'd get the "Login can only be changed 2 times every 7 days." message, but you'd still only be able to rename yourself once every seven days. Or am I missing something?

(Actually, thinking about it ... I think in order to make `USER_RENAME_LIMIT` configurable, you have to use the audits table to check the number of time the user has renamed in the last `USER_RENAME_LIMIT_DAYS` days, since the time of the most recent rename won't tell you how many other renames the user has done recently. So there's not much point in introducing the `renamed_at` column, since the calculation will be based on the audits table regardless. I think the description of the issue needs to be revisited — either get rid of the `renamed_at` column and rely on the audits table, or get rid of the ability to configure `USER_RENAME_LIMIT`.)

Reply:
I'd lean toward getting rid of `USER_RENAME_LIMIT` -- I don't really see us going with twice a week or three times every two weeks or something weird like that. Once per a configurable period of time seems fine.

Reply:
Ah, thanks for asking -- I meant to bring this up but totally forgot!